static/js/compose_actions.js

```diff
@@ -98,6 +107,18 @@ function show_box(msg_type, opts) {
     set_focus(msg_type, opts);
 }
 
+let togglable_message_type;
+
+function toggle_message_type(opts) {
+    togglable_message_type = togglable_message_type == "private" ? "stream" : "private";
+    compose_state.set_message_type(togglable_message_type);
+    show_box(togglable_message_type, opts);
+    update_placeholder_text();
+    // hm opts maybe also need to change? it seems visually fine but i don't like having to thread opts through
+    // the function calls like this, esp without rederiving them. also not a fan of the turning the event handler
+    // on and off. this feels like a jquery kind of issue -- i'll ask about best practices
```

Comment:
I'm new to production jquery, and a few things feel not great to me but I'm not sure which are worth fixing or what the best way to fix them are:

* Is having `togglable_message_type` a global variable okay? Should all other references to `message_type` reference the global variable, or is it okay to keep those separate?
* Am I missing updates in this function? I think I should be updating `opts` more but things look visually fine so maybe it's okay. I tried calling `start` but that resets several things and I want to change as few things as are necessary.

Reply:
@andersk not sure if you're the one to ask, but I'd love some feedback on the js I've written so far, since I'm still learning what jquery/zulip best practices look like.